include/common/core/arch_config.hpp

```diff
@@ -83,74 +87,114 @@ struct load_store_attr_t<msg_type::block_2d, gpu_arch::XeLpg>
           msg_type::block_2d,
           gpu_arch::XeLpg> {};
 
+template <gpu_arch arch_tag>
+inline constexpr bool arch_has_2d_load_store() {
+  using block2d_attr = load_store_attr_t<msg_type::block_2d, arch_tag>;
+  return block2d_attr::has_hw_block_2d;
+}
+
```

Comment:
Then maybe remove here
https://github.com/intel/neural-speed/blob/53093e5480475076764ff10637ca67eb5e65155b/include/common/core/common_types.hpp#L25-L30

Similar for `arch_has_xmx`